### PR TITLE
Add more sp api typings

### PR DESCRIPTION
**Pull request by jakeleventhal**
Adds typings for
- `getCatalogItem` query
- All report types
- `getReports` response and query
- Changes `getCatalogItem` to `any` which is more correct than the very incorrect type currently in place

### CODE REVIEW

File: lib/typings/operations/catalogItems.ts [hunk 1]
```diff
@@ -1,15 +1,17 @@
...
+  marketplaceIds: string[];
 }
 
 export interface GetCatalogItemPath {
   asin: string;
 }
 
 export interface GetCatalogItemResponse extends BaseResponse {
-  payload?: ListCatalogItem;
+  payload?: any;
```
Comment (jakeleventhal) [inline lib/typings/operations/catalogItems.ts:14]: This is a more correct type than `ListCatalogItem`

File: lib/typings/operations/reports.ts [hunk 2]
```diff
@@ -1,9 +1,214 @@
...
+  | "GET_B2B_PRODUCT_OPPORTUNITIES_RECOMMENDED_FOR_YOU"
+  | "GET_B2B_PRODUCT_OPPORTUNITIES_NOT_YET_ON_AMAZON";
+
+type RegulatoryComplianceReportType =
+  | "MARKETPLACE_ASIN_PAGE_VIEW_METRICS"
+  | "GET_EPR_MONTHLY_REPORTS"
+  | "GET_EPR_QUARTERLY_REPORTS"
+  | "GET_EPR_ANNUAL_REPORTS";
+
+export type ReportType =
```
Comment (jakeleventhal) [inline lib/typings/operations/reports.ts:181]: **bows**

File: lib/typings/operations/catalogItems.ts (hunk 1 above)
Comment (amz-tools) [reply lib/typings/operations/catalogItems.ts:14]: Good to have GetCatalogItemQuery updated to the new structure, the previous one was still from the deprecated v0 endpoint.

Comment (jakeleventhal) [reply lib/typings/operations/catalogItems.ts:14]: can this be punted to a separate issue? i was going to add it but the typings are really large for this

Comment (jakeleventhal) [reply lib/typings/operations/catalogItems.ts:14]: scope of this PR is already large

Comment (amz-tools) [reply lib/typings/operations/catalogItems.ts:14]: Yes, I think thats fine. The TS part would need some more major updates anyway, things that have been introduced in the meantime (i.e. endpoint param etc.) are also not included yet.

Comment (jakeleventhal) [reply lib/typings/operations/catalogItems.ts:14]: I'm not sure what the ask is here

**Pull request merged**